Android-Sample/app/src/main/java/com/opentok/qualitystats/sample/MainActivity.java

```diff
@@ -2,407 +2,205 @@
+import android.widget.TextView;
+
+import com.github.mikephil.charting.charts.LineChart;
+import com.github.mikephil.charting.components.XAxis;
+import com.github.mikephil.charting.components.YAxis;
+import com.github.mikephil.charting.data.Entry;
+import com.github.mikephil.charting.data.LineData;
+import com.github.mikephil.charting.data.LineDataSet;
+import com.github.mikephil.charting.formatter.ValueFormatter;
 import com.opentok.android.Publisher;
-import com.opentok.android.PublisherKit;
-import com.opentok.android.Session;
-import com.opentok.android.Stream;
-import com.opentok.android.Subscriber;
-import com.opentok.android.SubscriberKit;
-import com.opentok.android.SubscriberKit.VideoStatsListener;
-
-
-import pub.devrel.easypermissions.AfterPermissionGranted;
-import pub.devrel.easypermissions.AppSettingsDialog;
-import pub.devrel.easypermissions.EasyPermissions;
+import com.opentok.qualitystats.sample.models.NetworkQualityTestCallbackListener;
+import com.opentok.qualitystats.sample.models.NetworkQualityTestConfig;
+import com.opentok.qualitystats.sample.models.stats.CallbackQualityStats;
 
+import java.util.ArrayList;
 import java.util.List;
 
-public class MainActivity extends Activity implements
-        Session.SessionListener,
-        PublisherKit.PublisherListener,
-        SubscriberKit.SubscriberListener,
-        EasyPermissions.PermissionCallbacks{
-
-    private static final int RC_SETTINGS_SCREEN_PERM = 123;
-    private static final int RC_VIDEO_APP_PERM = 124;
-
-    private static final String LOGTAG = "quality-stats-demo";
-
-    private static final String SESSION_ID = "";
-    private static final String TOKEN = "";
-    private static final String APIKEY = "";
-
-    private static final int TEST_DURATION = 20; //test quality duration in seconds
-    private static final int TIME_WINDOW = 3; //3 seconds
-    private static final int TIME_VIDEO_TEST = 15; //time interval to check the video quality in seconds
-
-
-    private Session mSession;
-    private Publisher mPublisher;
-    private Subscriber mSubscriber;
-
-    private double mVideoPLRatio = 0.0;
-    private long mVideoBw = 0;
-
-    private double mAudioPLRatio = 0.0;
-    private long mAudioBw = 0;
-
-    private long mPrevVideoPacketsLost = 0;
-    private long mPrevVideoPacketsRcvd = 0;
-    private double mPrevVideoTimestamp = 0;
-    private long mPrevVideoBytes = 0;
-
-    private long mPrevAudioPacketsLost = 0;
-    private long mPrevAudioPacketsRcvd = 0;
-    private double mPrevAudioTimestamp = 0;
-    private long mPrevAudioBytes = 0;
-
-    private long mStartTestTime = 0;
-
-    private boolean audioOnly = false;
-
-    private Handler mHandler = new Handler();
-
-    private ProgressDialog mProgressDialog;
-    private AlertDialog dialog;
+public class MainActivity extends Activity {
+    static final String LOGTAG = "quality-stats-demo";
+    private static final String SESSION_ID = "YOUR_SESSION_ID";
```

Comment:
For the most distracted users, let's add some documentation on the README about this. By not being empty anymore the app will no longer alert the user and will try to connect for the duration of the timeout.